akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala

```diff
@@ -268,4 +277,44 @@ import akka.util.Timeout
    * INTERNAL API
    */
   @InternalApi private[akka] def messageAdapters: List[(Class[_], Any => T)] = _messageAdapters
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi private[akka] def setCurrentActorThread(): Unit = {
+    _currentActorThread match {
+      case OptionVal.None =>
+        _currentActorThread = OptionVal.Some(Thread.currentThread())
+      case OptionVal.Some(t) =>
+        throw new IllegalStateException(
```

Comment:
what scenario do do this? Is this a bug in our adapter?

Reply:
Yes, I think that must be a bug if that happens. Setting the thread twice, or not clearing it.